drivers/pcie/host/pcie.c

```diff
@@ -277,14 +281,25 @@ unsigned int pcie_alloc_irq(pcie_bdf_t bdf)
 {
 	unsigned int irq;
 	uint32_t data;
+#ifdef CONFIG_ACPI
+	const struct device *dev;
+#endif
 
 	data = pcie_conf_read(bdf, PCIE_CONF_INTR);
 	irq = PCIE_CONF_INTR_IRQ(data);
 
 	if (irq == PCIE_CONF_INTR_IRQ_NONE ||
 	    irq >= CONFIG_MAX_IRQ_LINES ||
 	    arch_irq_is_used(irq)) {
+#ifdef CONFIG_ACPI
+		dev = device_get_binding("acpi");
```

Comment:
Just move the `const struct device *dev` here. That way you don't need to have two #ifdef blocks in this function.